core/trino-spi/src/main/java/io/trino/spi/session/PropertyMetadata.java

```diff
@@ -26,35 +27,51 @@
 import static io.trino.spi.type.VarcharType.VARCHAR;
 import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
 import static java.lang.String.format;
+import static java.util.EnumSet.noneOf;
 import static java.util.Locale.ENGLISH;
 import static java.util.Objects.requireNonNull;
 import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toCollection;
 
 public final class PropertyMetadata<T>
 {
+    public static final EnumSet<Flag> NO_FLAGS = noneOf(Flag.class);
+
+    public static EnumSet<Flag> flags(Flag... flags)
+    {
+        return Arrays.stream(flags).collect(toCollection(() -> EnumSet.noneOf(Flag.class)));
+    }
+
+    public enum Flag
+    {
+        IGNORE_ON_CREATE_LIKE,
+        HIDDEN
+    }
+
     private final String name;
     private final String description;
     private final Type sqlType;
     private final Class<T> javaType;
     private final T defaultValue;
-    private final boolean hidden;
     private final Function<Object, T> decoder;
     private final Function<T, Object> encoder;
+    private final EnumSet<Flag> flags;
 
     public PropertyMetadata(
             String name,
             String description,
             Type sqlType,
             Class<T> javaType,
             T defaultValue,
-            boolean hidden,
+            EnumSet<Flag> flags,
```

Comment:
Use `Set<Flag>` and make a defensive copy to ensure that `PropertyMetadata` is immutable:
```java
this.flags = unmodifiableSet(EnumSet.copyOf(flags));